src/semgrep_agent/semgrep_app.py

```diff
@@ -124,6 +125,31 @@ def download_rules(self) -> Tuple[Path, int]:
         rules_path.write_text(rules)
         return rules_path, len(parsed["rules"])
 
+    def report_failure(self, error: SemgrepError) -> int:
+        """
+        Send semgrep cli non-zero exit code information to server
```

Comment:
🙏 thank you for the nice docstrings as usual. I gotta get back into this habit myself.